vaadin-checkbox-flow-demo/src/main/java/com/vaadin/flow/component/checkbox/demo/CheckboxGroupView.java

```diff
@@ -34,174 +47,223 @@
-        group.setEnabled(false);
+        // source-example-heading: Entity list
+        CheckboxGroup<Department> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setLabel("Department");
+        List<Department> departmentList = getDepartments();
+        checkboxGroup.setItems(departmentList);
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
         // end-source-example
 
-        group.setId("checkbox-group-disabled");
-
-        addCard("Disabled checkbox group", group);
+        addCard("Entity list", checkboxGroup);
     }
 
-    private void addDisabledItems() {
-
-        Div valueInfo = new Div();
+    private void valueChangeEvent() {
         // begin-source-example
-        // source-example-heading: Checkbox group with item enabled
-        // provider
-        CheckboxGroup<String> group = new CheckboxGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.setItemEnabledProvider(item -> !"bar".equals(item));
+        // source-example-heading: Value change event
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setLabel("Label");
+        checkboxGroup.setItems("Option one", "Option two", "Option three");
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+
+        Div value = new Div();
+        value.setText("Select a value");
+        checkboxGroup.addValueChangeListener(event -> {
+            if (event.getValue() == null) {
+                value.setText("No option selected");
+            } else {
+                value.setText("Selected: " + event.getValue());
+            }
+        });
         // end-source-example
 
-        group.addValueChangeListener(
-                event -> valueInfo.setText(toString(group.getValue())));
-
-        group.setId("checkbox-group-disabled-items");
-        valueInfo.setId("checkbox-group-disabled-items-info");
-
-        addCard("Checkbox group with item enabled provider", group, valueInfo);
+        addCard("Value change event", checkboxGroup, value);
     }
 
-    private void addReadOnlyGroup() {
+    private void indeterminateCheckbox() {
         // begin-source-example
-        // source-example-heading: Read-only checkbox group
-        Div valueInfo = new Div();
+        // source-example-heading: Indeterminate checkbox
+        Checkbox checkbox = new Checkbox("Select all");
+        checkbox.setIndeterminate(true);
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        Set<String> items = new LinkedHashSet<>(
+                Arrays.asList("Option one", "Option two"));
+        checkboxGroup.setItems(items);
+        List<CheckboxGroup> checkboxList = new ArrayList<>();
+        checkboxList.add(checkboxGroup);
+        checkboxGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
+        checkboxGroup.addValueChangeListener(
+                checkboxGroupSetComponentValueChangeEvent -> {
```

Comment:
This event name is quite verbose "checkboxGroupSetComponentValueChangeEvent". Just "event" should be enough.

Reply:
Done.